src/main/java/org/aksw/simba/lemming/metrics/single/MaxVertexDegreeMetric.java

```diff
@@ -13,20 +19,261 @@
+	 *            that can be used.
+	 * @return UpdatableMetricResult object with updated candidate set and metric
+	 *         values.
+	 */
+	@Override
+	public UpdatableMetricResult applyUpdatable(ColouredGraph graph, UpdatableMetricResult previousResult) {
+		MaxVertexDegreeMetricResult metricResultTempObj = new MaxVertexDegreeMetricResult(getName(), 0.0);
+
+		if (previousResult instanceof MaxVertexDegreeMetricResult) {
+			// Set previously stored maps
+			metricResultTempObj
+					.setmMapCandidatesMetric(((MaxVertexDegreeMetricResult) previousResult).getmMapCandidatesMetric());
+			metricResultTempObj.setmMapCandidatesMetricValues(
+					((MaxVertexDegreeMetricResult) previousResult).getmMapCandidatesMetricValues());
+		}
+
+		return metricResultTempObj;
+	}
+
+	/**
+	 * The method checks if we need to compute in degree or out-degree and then
+	 * calls the metricComputationMaxDegree with correct parameters.
+	 * 
+	 * @param triple
+	 *            - edge on which graph operation is performed.
+	 * @param metric
+	 *            - input metric which needs to be computed.
+	 * @param graph
+	 *            - input graph.
+	 * @param graphOperation
+	 *            - boolean value indicating graph operation. ("true" for adding an
+	 *            edge and "false" for removing an edge)
+	 * @param previousResult
+	 *            - UpdatableMetricResult object containing the previous computed
+	 *            results.
+	 * @return
+	 */
+	@Override
+	public UpdatableMetricResult update(TripleBaseSingleID triple, ColouredGraph graph, boolean graphOperation,
+			UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+		UpdatableMetricResult newMetricResult;
+
+		switch (getName()) {
+		case "maxInDegree":
+
+			if (graphOperation) { // graphOperation is true then add an edge otherwise its remove an edge
+				newMetricResult = metricComputationMaxDegree(graph, GRAPHOPERATION.AddAnEdgeIndegree, DIRECTION.in,
+						triple.headId, triple, 1, previousResult, mVertexDegrees);
+			} else {
+				newMetricResult = metricComputationMaxDegree(graph, GRAPHOPERATION.RemoveAnEdgeIndegree, DIRECTION.in,
+						triple.headId, triple, -1, previousResult, mVertexDegrees);
+			}
+
+			break;
+		// Logic to check if maxDegreeMetric call is required for maxInDegree, defined
+		// in method : metricComputationMaxDegree
+
+		case "maxOutDegree":
+			if (graphOperation) {
+				newMetricResult = metricComputationMaxDegree(graph, GRAPHOPERATION.AddAnEdgeOutdegree, DIRECTION.out,
+						triple.tailId, triple, 1, previousResult, mVertexDegrees);
+			} else {
+				newMetricResult = metricComputationMaxDegree(graph, GRAPHOPERATION.RemoveAnEdgeOutdegree, DIRECTION.out,
+						triple.tailId, triple, -1, previousResult, mVertexDegrees);
+			}
+			break;
+
+		default:// If metric is other than maxInDegree and maxOutDegree then apply the metric
+			newMetricResult = applyUpdatable(graph, previousResult);
+		}
```

Comment:
I do not really understand why you rely on the two names. You should rely on the direction as the apply method shows.

Reply:
Done. I have changed it, now using direction instead of metric names.